Create rhai docs pages (#700)
apollographql/apollo-mcp-server: docs/source/rhai-functions.mdx

| `Regex::matches(string, pattern)` | Returns an array of all matching substrings. Returns an empty array if there are no matches. Throws on invalid pattern. | `Regex::matches("abc 123 def 456", "\\d+")` |

```rhai
// Check if a string contains digits
let has_numbers = Regex::is_match("order-123", "\\d+");
// has_numbers = true

// Replace all matches
let result = Regex::replace("foo bar foo", "foo", "baz");
// result = "baz bar baz"

// Use capture groups to reformat a date
let reformatted = Regex::replace("2025-01-15", "(\\d{4})-(\\d{2})-(\\d{2})", "$2/$3/$1");
// reformatted = "01/15/2025"

// Use named capture groups
let swapped = Regex::replace("John Smith", "(?P<first>\\w+) (?P<last>\\w+)", "$last, $first");
// swapped = "Smith, John"

// Find all matches
let numbers = Regex::matches("abc 123 def 456", "\\d+");
// numbers = ["123", "456"]
```

## ErrorCode

Constants for structured error responses when using `throw` in lifecycle hooks. See [Lifecycle Hooks](/apollo-mcp-server/rhai-lifecycle#error-handling-with-throw) for details on error handling.

| Constant | Description |
|----------|-------------|

> | `ErrorCode::INVALID_REQUEST` | Indicates a client error, such as a missing header or invalid input. |

Reviewer: ```suggestion
| `ErrorCode::INVALID_REQUEST` | Indicates a client error, like a missing header or invalid input. |
```

> | `ErrorCode::INTERNAL_ERROR` | Indicates a server-side error. This is the default when no code is specified. |

Reviewer: ```suggestion
| `ErrorCode::INTERNAL_ERROR` | Indicates a server-side error. This is the default when no other error code is specified. |
```
